spec/requests/vms_spec.rb

```diff
@@ -82,6 +82,43 @@ def update_raw_power_state(state, *vms)
                                    [{"vendor" => "openstack"},
                                     {"vendor" => "openstack"}])
     end
+
+    def add_hardware_and_os_to_vms
+      [vm, vm1, vm2, vm_openstack, vm_openstack1, vm_openstack2].each do |vm_record|
+        cs = ComputerSystem.create
+        os = OperatingSystem.create(:name => "linux", :vm_or_template => vm_record, :computer_system => cs)
+        FactoryBot.create(:hardware, :vm_or_template => vm_record, :host => host)
```

Comment:
Added this because there is a quirk with the API where it doesn't render nested keys if the relationship is `nil`, and this was a way to ensure that the records were being loaded as expected.